extensions/positron-python/src/client/positron/extension.ts

```diff
@@ -71,6 +74,146 @@ export async function activatePositron(serviceContainer: IServiceContainer): Pro
         disposables.push(
             vscode.commands.registerCommand('python.getMinimumPythonVersion', (): string => MINIMUM_PYTHON_VERSION.raw),
         );
+
+        // Register run application commands.
+        // TODO: Once we have contexts defined for the various application frameworks (e.g. isShinyApp),
+        //       We could have a single "Run Python App in Terminal" command with behavior conditional
+        //       on the context value.
+        disposables.push(
+            vscode.commands.registerCommand('python.runDashApp', async () => {
+                const runAppApi = await getPositronRunAppApi();
+                await runAppApi.runApplication({
+                    name: 'Dash',
+                    getTerminalOptions(runtime, document, port, urlPrefix) {
+                        const env: RunAppTerminalOptions['env'] = {};
+                        if (port) {
+                            env.DASH_PORT = port;
+                        }
+                        if (urlPrefix) {
+                            env.DASH_URL_PREFIX = urlPrefix;
+                        }
+                        return {
+                            commandLine: [runtime.runtimePath, document.uri.fsPath].join(' '),
+                            env,
+                        };
+                    },
+                });
+            }),
+
+            vscode.commands.registerCommand('python.runFastAPIApp', async () => {
+                const runAppApi = await getPositronRunAppApi();
+                await runAppApi.runApplication({
+                    name: 'FastAPI',
+                    async getTerminalOptions(runtime, document, port, urlPrefix) {
+                        const appName = await getAppName(document, 'FastAPI');
+                        if (!appName) {
+                            return undefined;
+                        }
+                        const args = [
```

Comment:
actually, can we avoid using the `path:appName` and just run `fastapi dev path/app.py` (or `fastapi run path/app.py`)?

Reply:
Unfortunately, fastapi-cli may not be installed, but I've updated to use it if it's installed otherwise default back to uvicorn as it was.